google/cloud/security/inventory/pipelines/base_pipeline.py

```diff
@@ -54,6 +54,7 @@ def __init__(self, cycle_timestamp, global_configs, api_client, dao):
     @abc.abstractmethod
     def run(self):
         """Runs the pipeline."""
+        # TODO: should this raise an error?
```

Comment:
Since this is an abstract method of the class, subclasses have to define their own implementation of foo or they will fail with TypeError at runtime:

>>> import abc
>>> class test(object):
...   __metaclass__ = abc.ABCMeta
...   @abc.abstractmethod
...   def run(self):
...     pass
... 
>>> class subtest(test):
...   def other(self):
...     pass
... 
>>> s = subtest()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Can't instantiate abstract class subtest with abstract methods run